### DISCUSSION TITLE
Run node-agent locally?

### DISCUSSION THREAD

**Discussion by mame-leadec** [vmware-tanzu/velero#7754] [Community support Q&A] [answered]
Hello,

I have experimented a bit with the function to run the Velero server locally. [run-locally](https://velero.io/docs/v1.13/run-locally/)

I would also like to use the CSI Snapshot Data Movement locally. But as far as I understand it, the Node Agent is responsible for this and it only runs on the cluster.

Does anyone know if there is a way to run the node agent locally?

**Comment by sseago** [comment, marked as answer]
There's no way to run that locally -- the node agent must run on the same node as the pod which mounts the volume being copied from or to, since it otherwise won't have access to the node's mount point for the volume.

**Comment by mame-leadec** [comment]
Makes sense, thank you for the clarification. 

So it will be just not possible to run Velero Server in that special case. (locally)
